### PR TITLE
Fix mono & gcc build issues

### LINKED ISSUE #61: mono support
Hi,

I attempted to use this with a mono enabled godot, however it broke on generating the glue code.  Am I missing anything here?  Should it work..?

I have compiled from godot git before, however I have not attempted to add a new module in.  Thanks.

### LINKED ISSUE #67: Broken build on OSX
Using current godot master (commit 24e1039eb6fe32115e8d1a62a84965e9be19a2ed) at the time of this writing, as well as godot_voxel master. Getting this: 
image

### PR DESCRIPTION
**Pull request by tinmanjuggernaut**
@lordee Can you confirm that this PR fixes the warnings?

Errors already fixed in 1ad71c5.

Fixes #61 
Fixes #67

### CODE REVIEW

**Comment by tinmanjuggernaut** [comment]
I left these two warnings there:

`modules/voxel/streams/voxel_stream_region_files.cpp:403:13: warning: 'bool from_json_varray(Array, Vector3i&)' defined but not used`

`modules/voxel/streams/voxel_stream_region_files.cpp:380:14: warning: 'Array to_varray(const Vector3i&)' defined but not used`

**Comment by lordee** [comment]
Hi,

This is the gist for this PR:
https://gist.github.com/lordee/6b735d770f4bbdf126743984682270bd

**Comment by tinmanjuggernaut** [comment]
This PR clears all mono, gcc errors and warnings, except for the forward declarations mentioned in #67 and the unused function warning mentioned below. Includes OSX fixes from #67 @whpthomas.

```
modules/voxel/streams/voxel_stream_region_files.cpp:403:13: warning: 'bool from_json_varray(Array, Vector3i&)' defined but not used [-Wunused-function]
 static bool from_json_varray(Array a, Vector3i &v) {
             ^~~~~~~~~~~~~~~~
modules/voxel/streams/voxel_stream_region_files.cpp:380:14: warning: 'Array to_varray(const Vector3i&)' defined but not used [-Wunused-function]
 static Array to_varray(const Vector3i &v) {
              ^~~~~~~~~
```

**Comment by tinmanjuggernaut** [comment]
This could be merged as is, assuming @Zylann will replace the noted function in #67. That change should close #67. Alternatively I can add voxel_map.h to the specified files in this PR, in which case, this can close #67.

File: terrain/voxel_block.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@ static Ref<ConcavePolygonShape> create_concave_polygon_shape(Array surface_array
 	ERR_FAIL_COND_V(indices.size() % 3 != 0, Ref<ConcavePolygonShape>());
 
 	int face_points_count = indices.size();
-	int face_count = face_points_count / 3;
+	//int face_count = face_points_count / 3;
```
Comment (Zylann) [inline terrain/voxel_block.cpp]: Just remove this

**Comment by tinmanjuggernaut** [comment]
Ok, this addresses all the non-unused-function warnings and errors for linux/gcc standard and mono, and the errors for osx (apparently there are many warnings). This also works with the 3.1 build, given the pending PR I will submit.

File: terrain/voxel_block.cpp (hunk 1 above)
Comment (tinmanjuggernaut) [reply terrain/voxel_block.cpp]: Done

**Comment by Zylann** [comment]
Thanks!

**Pull request merged**